policy/release/tasks.rego

```diff
@@ -169,10 +169,31 @@ _missing_tasks(required_tasks) := {task |
 	some att in lib.pipelinerun_attestations
 	count(tkn.tasks(att)) > 0
 
-	some task in required_tasks
-	not task in tkn.tasks_names(att)
+	some required_task in required_tasks
+	some task in _any_missing(required_task, tkn.tasks_names(att))
 }
 
+_any_missing(required, tasks) := missing if {
+	# one of required tasks is required
+	is_array(required)
```

Comment:
Could use a helper method to avoid having to implement the logic for both cases, e.g.
```rego
_any_missing(required, tasks) := missing if {
  req := setfy(required)
  tsk := setfy(tasks)
  ...
}

setfy(o) := s if {
  s := {v | some v in o}
} else := {o}
```

This could maybe reside in `policy/lib/set_helpers.rego`.

Reply:
Turns out we already have that helper there :)

Reply:
I think it would've been nice to unify the two use cases (array vs non-array) which is what I was trying to drive at with this comment. But anyways, the current approach works just as well :wink: